### PR TITLE
SketchUnmap macro creation

### PR DESCRIPTION
**Pull request by 0penBrain**
Thank you for creating a pull request to contribute to FreeCAD-macros!
To integrate your macro please make sure the following steps are complete:

- [ ] Please check this box if you're not submitting a new macro.   
- [X] Are you submitting a new macro ?  
- [X] Have you followed the ['How to submit a macro'](README.md#how-to-submit-a-macro) section of the README.md ?  
- [X] Your macro has a [Description](README.md#macro-description) in its header.  
- [X] Your macro has a [CamelCase name](README.md#camelcase-macro-name).  
- [X] Your macro is named [appropriately](README.md#macro-name-specifics).  
- [X] Your macro contains a [Metadata section](README.md#macro-metadata) that immediately follows the header description.  
- [X] Your macro is Python3/Qt5 compliant and tested on the latest FreeCAD stable and development releases.  
- [X] You're including documentation on how your macro works (bonus: screenshots and/or video on the Wiki)  
- [X] Commit message is [well-written](https://chris.beams.io/posts/git-commit/)  
- [ ] Commit message is titled in the following way `[MacroName] - Short description`.  

And please remember to update the Wiki with the features added or changed once this PR is merged.  
**Note**: If you don't have wiki access, then please mention your contribution on the [0.19 Changelog Forum Thread](https://forum.freecadweb.org/viewtopic.php?f=10&t=34586).

---

### CODE REVIEW

File: Sketcher/SketchUnmap.FCMacro [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+_0Vec_ = FreeCAD.Vector(0, 0, 0) #Shortener for null vector
+_ZVec_ = FreeCAD.Vector(0, 0, 1) #Shortener for Z axis
+
+if __dbg__:  ##Clear report view in debug mode
+    FreeCADGui.getMainWindow().findChild(QtGui.QTextEdit, "Report view").clear()
+
+cslM("Starting SketchUnmap macro")
+cslD("Checking selection")
+
+if (len(Gui.Selection.getSelection()) != 1 or str(Gui.Selection.getSelection()[0]) != '<Sketcher::SketchObject>'): ##If selection is wrong, print error message and exit
```
Comment (luzpaz) [inline Sketcher/SketchUnmap.FCMacro]: Please put comment above the code block in this case

File: Sketcher/SketchUnmap.FCMacro [hunk 2]
```diff
@@ -0,0 +1,117 @@
+#!/usr/bin/python
+#####################################
+# Copyright (c) openBrain 2019
+# Licensed under LGPL v2
+#
+# This macro will unmap a sketch from eg. a face and make its placement absolute in the body. It proposes 3 options (forcely the first if sketch not in a body) :
```
Comment (luzpaz) [inline Sketcher/SketchUnmap.FCMacro]: Please mention FreeCAD in this this description like `This FreeCAD macro will unmap...`

File: Sketcher/SketchUnmap.FCMacro [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+# * "Raw" mode => the sketch placement is made absolute in the body referential, nothing more
+# * "DP@Face mode" => a datum plane is created where the mapping face is, then the sketch is attached to it respecting its attachment offset
+# * "DP@Sketch" mode => a datum place is created where the sketch is (including attachment offset), then the sketch is attached to its origin
+#
+# Main use of the macro is to make design more robust to topological naming issue
+#
+# Version history :
+# *0.5 : beta release
```
Comment (luzpaz) [inline Sketcher/SketchUnmap.FCMacro]: please consider adding the latest revision from recent to oldest instead

File: Sketcher/SketchUnmap.FCMacro [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+    __Date__ = '2019-06-10'
+    __License__ = 'LGPL v2'
+    __Web__ = 'https://www.freecadweb.org/wiki/Macro_SketchUnmap'
+    __Wiki__ = 'https://www.freecadweb.org/wiki/Macro_SketchUnmap'
+    __Icon__ = ''
+    __Help__ = 'Select the sketch to unmap (eg. in the tree view) then run the macro'
+    __Status__ = 'Beta'
+    __Requires__ = 'FreeCAD >= 0.17'
+    __Communication__ = 'https://forum.freecadweb.org/viewtopic.php?f=22&t=36078'
+    __Files__ = ''
```
Comment (luzpaz) [inline Sketcher/SketchUnmap.FCMacro]: Files associated with this macro need to be added here.

File: Sketcher/SketchUnmap.FCMacro (hunk 1 above)
Comment (0penBrain) [reply Sketcher/SketchUnmap.FCMacro]: Sorry but could you be more precise on what you mean by "in this case" ? In which case(s) do you expect the comment to be above the code block ?

File: Sketcher/SketchUnmap.FCMacro (hunk 2 above)
Comment (0penBrain) [reply Sketcher/SketchUnmap.FCMacro]: OK. Why not precise it on the readme of the repo ? ;)

File: Sketcher/SketchUnmap.FCMacro (hunk 3 above)
Comment (0penBrain) [reply Sketcher/SketchUnmap.FCMacro]: OK.

File: Sketcher/SketchUnmap.FCMacro (hunk 4 above)
Comment (0penBrain) [reply Sketcher/SketchUnmap.FCMacro]: I don't have any ATM. I kept property for potential future icon. ;) Do you prefer I remove the line while there is no linked file ?

File: Sketcher/SketchUnmap.FCMacro (hunk 1 above)
Comment (luzpaz) [reply Sketcher/SketchUnmap.FCMacro]: I think when the comment extends to far right past the margin. @galou are we planning to use `pep8` to keep coding style to a certain standard.

File: Sketcher/SketchUnmap.FCMacro (hunk 2 above)
Comment (luzpaz) [reply Sketcher/SketchUnmap.FCMacro]: Yea, good idea. Needs to go there.

File: Sketcher/SketchUnmap.FCMacro (hunk 4 above)
Comment (luzpaz) [reply Sketcher/SketchUnmap.FCMacro]: @0penBrain ah, apologies, you were aware of this and you raise a good question.   
@galou what do think?

File: Sketcher/SketchUnmap.FCMacro (hunk 1 above)
Comment (0penBrain) [reply Sketcher/SketchUnmap.FCMacro]: IMHO, strictly applying PEP8 could refrain users to contribute. ;)

**Comment by 0penBrain** [comment]
@luzpaz : all remarks took into account. Tell me if more needed. ;)

File: Sketcher/SketchUnmap.FCMacro (hunk 4 above)
Comment (0penBrain) [reply Sketcher/SketchUnmap.FCMacro]: @luzpaz no problem. :) Actually that was a guess. There is also the __Icon__ metadata that I kept blank. I can't determine from the repo readme if icon file shall also be included in the __Files__ metadata ??? So I kept it blank to think about it. I commented out this line so everything is clear (hopefully). :)

Comment (galou) [reply Sketcher/SketchUnmap.FCMacro]: The `__Files__` metada should exits even if  it's empty. We actually do not do anything right now with the `__Icon__` metadata, as far as I know.

Comment (0penBrain) [reply Sketcher/SketchUnmap.FCMacro]: @galou thanks for clarifying. Could this be indicated in the README.md of the repo with something like "Please keep all metadata even if empty" ?

**Pull request merged**